File: 프로그래머스/unrated/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.cs
using System;
using System.Text;

public class Solution {
    public string solution(string my_string, string overwrite_string, int s) 
    {
        var sb = new StringBuilder();
        
        sb.Append(my_string.Substring(0, s));
        sb.Append(overwrite_string);
        
        int end = s + overwrite_string.Length;
        sb.Append(my_string.Substring(end, my_string.Length - end));
        
        return sb.ToString();
    }
}

File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.cpp
#include <string>
#include <vector>
#include <sstream>

using namespace std;

string solution(string my_string, string overwrite_string, int s) {
    string answer = "";
    
    std::ostringstream oss;
    
    int i = 0;
    for (; i < s; ++i)
        oss << my_string[i];
    
    oss << overwrite_string;
    i += overwrite_string.length();
    
    for (; i < my_string.length(); ++i)
        oss << my_string[i];
        
    return oss.str();
}